src/hardware/sblaster.cpp

```diff
@@ -294,6 +308,157 @@ static void InitializeSpeakerState()
 	}
 }
 
+static void configure_filters(Section_prop* config)
+{
+	auto get_filter_params = [&](const char * conf) {
+		static const std::map<SB_TYPES, FilterType> sb_type_to_filter_type_map = {
+				{SBT_NONE, FilterType::None},
+				{SBT_1, FilterType::SB1},
+				{SBT_2, FilterType::SB2},
+				{SBT_PRO1, FilterType::SBPro1},
+				{SBT_PRO2, FilterType::SBPro2},
+				{SBT_16, FilterType::SB16},
+				{SBT_GB, FilterType::None},
+		};
+
+		static const std::map<std::string, FilterType> filter_map = {
+				{"none", FilterType::None},
+				{"sb1", FilterType::SB1},
+				{"sb2", FilterType::SB2},
+				{"sbpro1", FilterType::SBPro1},
+				{"sbpro2", FilterType::SBPro2},
+				{"sb16", FilterType::SB16},
+		};
+		assert(conf);
+		const auto filter_prefs = split(config->Get_string(conf));
+		const auto filter = filter_prefs.empty() ? "auto" : filter_prefs[0];
+
+		FilterType filter_type = FilterType::None;
+		if (filter == "auto") {
+			auto it = sb_type_to_filter_type_map.find(sb.type);
+			if (it != sb_type_to_filter_type_map.end())
+				filter_type = it->second;
+		} else {
+			auto it = filter_map.find(filter);
+			if (it != filter_map.end())
+				filter_type = it->second;
+		}
+
+		FilterState filter_state = FilterState::Off;
+		if (filter_type != FilterType::None) {
+			if (filter_prefs.size() > 1 && filter_prefs[1] == "always_on") {
+				filter_state = FilterState::ForcedOn;
+			} else {
+				filter_state = FilterState::On;
+			}
+		}
+		return std::make_tuple(filter_type, filter_state);
+	};
+	using std::tie;
+	tie(sb.sb_filter_type, sb.sb_filter_state) = get_filter_params("sb_filter");
+	tie(sb.opl_filter_type, std::ignore) = get_filter_params("opl_filter");
+}
+
+static void log_filter_config(const char *output_type, FilterType filter)
```

Comment:
Suggest setting the FilterType to `const FilterType` (we're only reading the `filter` variable and not using it as a temporary).

Reply:
Yeah, I just forget sometimes. We already have tons of linters, static analysers and whatnot, would it be possible to emit warnings for these cases where a function arg or local var could be made const? That would be quite handy!

Reply:
> Yeah, I just forget sometimes. We already have tons of linters, static analysers and whatnot, would it be possible to emit warnings for these cases where a function arg or local var could be made const? That would be quite handy!

No worries - all part of having extra eyes; we're all helping :+1:

I'm not aware of any warning flags or tools (not even clang's `tidy`) that have this feature. My guess is to do it right, you'd almost need to compile the code to determine the read-only-ness of a variable.

But yeah;  I'm also a fan of `const` because it captures the author's intent using the Law of Demeter: _"if this value doesn't need to be modified, then don't allow it to be changed beyond this point"_, and the compiler gives us hard proof of that.

Apparently, it's reported to help compilers the same way: it's free to use whatever tricks it has with const value to speed up operations knowing it'll never change via some other side-effect (but I've never benchmarked it to prove it).

Hey.. maybe Scott Meyer's is working on a bonus effective C++ warning with this: `-wextra-effc++`?  j/k,, @shermp :sweat_smile: